Add data lab (#9)
open-education-hub/operating-systems: content/chapters/data/lab/content/working-memory.md

The address for `a` is automatically generated by going through multiple layers of abstractions, but the size is spelled out indirectly by the programmer by using the keyword `int`.
Another aspect is represented by the access rights for a specific memory area.
In our example, `a` is defined as being plain mutable, however, it is possible to declare constant variables which are stored in memory location with no writing rights.

Using the above information, the compiler and the operating system co-work to allocate memory that can represent the contents of the variable.

No matter what sort of language you are using, statically or dynamically typed, a variable is always described by the **(address, size, access rights)** triplet.
By using this triplet the content of a variable is stored, retrieved or rewritten.

#### Practice

Navigate to the `support/memory-access/` directory.
Inspect the `mem_access.c` source file.

1. Describe each variable by completing its **(address, size, access rights)** tuple.

1. Try to modify the `ca`,`cp` and `cp2` variables by assigning some other value to them.
   Explain the behavior.

#### Quiz

> [Quiz](quiz/memory-access.md)

Reviewer: ```suggestion
[Quiz](../quiz/memory-access.md)
```

### Memory Protection

> Memory contents (both code an data) is separated into sections or zones.

Author: ```suggestion
Memory contents (both code and data) are separated into sections or zones.
```
